Did the whole backend thing.

WombatusMaximus opened the pull request:
Database stuff through EF6, searches, getters, updaters. Everything except for users. Right now there is a nameless dummy with ID 1

WombatusMaximus committed:
Did the whole backend thing.
Database stuff through EF6, searches, getters, updaters. Everything except for users. Right now there is a nameless dummy with ID 1

momoadept commented on TeaDiary.business/Interfaces/ITeaRepository.cs:
```diff
@@ -0,0 +1,16 @@
+    public interface ITeaRepository
+    {
+        IEnumerable<Tea> GetAll(int userId);
+        Tea GetById(int teaId);
```
This doesn't have user restrictions. Maybe it should?

WombatusMaximus replied:
Implemented

momoadept commented on TeaDiary.business/Interfaces/IUserProvider.cs:
```diff
@@ -0,0 +1,10 @@
+
+namespace TeaDiary.business.Interfaces
+{
+    public interface IUserProvider
```
I have no idea what specific user you're talking about

WombatusMaximus replied:
Changed name for both interface as well as implementation. Will look through the code before commiting to get sure everything is ok

momoadept commented on TeaDiary.business/Values/ITeaGetter.cs:
```diff
@@ -0,0 +1,14 @@
+    public interface ITeaGetter
+    {
+        IList<Tea> GetAll();
+        Tea GetByID(int id);
```
`ID` should be `Id`

WombatusMaximus replied:
Should it be `ID` in tea and user models as well?

momoadept replied:
It should always be `Id`. Consider it to be a word and not an acronym.

WombatusMaximus replied:
I hope I made it right everywhere

momoadept commented on TeaDiary.business/Values/TeaGetter.cs:
```diff
@@ -0,0 +1,38 @@
+{
+    public class TeaGetter : ITeaGetter
+    {
+        private readonly ITeaRepository values;
```
`values` is not a good name

WombatusMaximus replied:
Changed to `teasRepository`

momoadept commented on TeaDiary.data_access/TeaDiaryContext.cs:
```diff
@@ -0,0 +1,21 @@
+
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Tea>().ToTable("Values");
```
Why isn't the table named "Teas"?

momoadept replied:
I think I know why. You have extra projects in the file system. You should remove them.

WombatusMaximus replied:
Yeah, was readding dataaccess because of wrong project template

WombatusMaximus replied:
Deleted excessive one. I hope so at least

momoadept commented on TeaDiary.dataaccess/Migrations/201912191154492_initial.cs:
```diff
@@ -0,0 +1,32 @@
+            DropForeignKey("dbo.Teas", "UserId", "dbo.Users");
+            DropIndex("dbo.Teas", new[] { "UserId" });
+            DropTable("dbo.Users");
+        }
```
This migration doesn't look right to me

momoadept commented on TeaDiary.dataaccess/Repositories/TeaRepository.cs:
```diff
@@ -0,0 +1,98 @@
+            {
+                throw new Exception();
+            }
+            else 
```
Always use `{}` even if there is one line of code

momoadept commented on TeaDiary.dataaccess/Repositories/TeaRepository.cs:
```diff
@@ -0,0 +1,98 @@
+            if (tea.ID == null)
+            {
+                return false;
+            }
```
Unless it is a part of if-else block, leave an empty line after `}`

WombatusMaximus replied:
Got it

momoadept commented on TeaDiary.dataaccess/Repositories/UserProvider.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace TeaDiary.dataaccess.Repositories
+{
+    public class UserProvider: IUserProvider
```
This should be in API project

WombatusMaximus replied:
Moved

momoadept commented on TeaDiary.domain/Models/Tea.cs:
```diff
@@ -4,8 +4,11 @@ namespace TeaDiary.domain.Models
 {
     public class Tea:Resource
     {
+        //public User User { get; set; }
```
Why no navigation property?

WombatusMaximus replied:
Cause didn't understand properly how it works. Learned it out. Fixed

momoadept commented on TeaDiary.domain/Models/User.cs:
```diff
@@ -9,7 +9,7 @@ public class User: Resource
         public string PasswordHash { get; set; }
         public string EmailAdress { get; set; }
-        public IEnumerable<Tea> Teas { get; set; }
+        public IList<int> TeaIDs { get; set; }
```
That's not how you model 'one to many' in DB. Your 'many' end should a foreign id (`tea.UserId`). Your 'one' end shouldn't have any foreign keys.

WombatusMaximus replied:
Made `IEnumerable<tea>` back instead of `IList<int>`

momoadept commented on TeaDiary/App_Start/AutofacConfig.cs:
```diff
@@ -9,6 +13,12 @@ public static void AddApplicationServices(this ContainerBuilder builder)
             //Register CONTROLLER as default INTERFACE realization
```
I think you mean INTERFACE and IMPLEMENTATION. Also, the correct word for "realization" is "implementation"

WombatusMaximus replied:
Was guessing how to call it properly. Thanks!

momoadept reviewed (changes requested):
8/10 would review again

WombatusMaximus committed:
Refactored backend code a bit using advises from momoadept.
Added User database, linked it with Tea DB. For some reason every time Seed runs new copy of test data is added instead of updating. Need momoadept's help with that

WombatusMaximus committed:
Fixed seed update on migrations. Changed CUP to fit database

momoadept commented on TeaDiary.dataaccess/Migrations/Configuration.cs:
```diff
@@ -27,21 +27,28 @@ protected override void Seed(TeaDiaryContext context)
+            context.Users.AddOrUpdate(initialUser);
+            context.Database.ExecuteSqlCommand("SET IDENTITY_INSERT USERS ON");
+            context.SaveChanges();
+            context.Database.ExecuteSqlCommand("SET IDENTITY_INSERT USERS OFF");
```
This doesn't work, don't keep it

WombatusMaximus replied:
What exactly should I delete?

momoadept replied:
`SET IDENTITY_INSERT` stuff

momoadept commented on TeaDiary.dataaccess/Migrations/Configuration.cs:
```diff
@@ -61,10 +69,14 @@ protected override void Seed(TeaDiaryContext context)
+            foreach (Tea initialTea in initialTeas)
+            {
+                context.Teas.AddOrUpdate(initialTea);
+            }
```
Why don't you just pass the array as you did before?

WombatusMaximus replied:
Well, I was experimenting with .Add which couldn't take array as param and forgot to change it back

momoadept replied:
Dew it

WombatusMaximus committed:
Minor configuraion refactoring
